pulpcore/pulpcore/app/serializers/repository.py

```diff
@@ -223,11 +245,9 @@ class DistributionSerializer(ModelSerializer):
                     'served as defined by this distribution'),
         queryset=models.Publisher.objects.all(),
     )
-    publication = serializers.HyperlinkedRelatedField(
+    publication = HyperlinkedRelatedPublicationField(
```

Comment:
We may not need to add a new publication field. Instead, we can just set the queryset to exclude complete=false here and use the vanilla HyperlinkedRelatedField

Reply:
ack.

Reply:
Actually I think a specific hyperlink here serves both reusability and documentation of intention. But I'm also cool with the vanilla hyperlink; just posted a revision to see how it feels the other way.

Reply:
I can see the benefit of creating a custom field, but I think its helpful to keep these explicit rather than create classes that hide the details. If we follow this pattern, we would have a custom field class for every relationship, which might be easier to use, but makes the serializers more opaque. I strongly prefer vanilla, especially in this case, since there is only one Publication relation.